google/cloud/pubsub_v1/subscriber/message.py

```diff
@@ -281,16 +316,74 @@ def modify_ack_deadline(self, seconds: int) -> None:
                 against.
         """
         self._request_queue.put(
-            requests.ModAckRequest(ack_id=self._ack_id, seconds=seconds)
+            requests.ModAckRequest(ack_id=self._ack_id, seconds=seconds, future=None)
         )
 
+    def modify_ack_deadline_with_response(self, seconds: int) -> "futures.Future":
+        """Resets the deadline for acknowledgement and returns the response
+        status via a future.
+
+        New deadline will be the given value of seconds from now.
+
+        The default implementation handles automatically modacking received messages for you;
+        you should not need to manually deal with setting ack deadlines. The exception case is
+        if you are implementing your own custom subclass of
+        :class:`~.pubsub_v1.subcriber._consumer.Consumer`.
+
+        Args:
+            seconds:
+                The number of seconds to set the lease deadline to. This should be
+                between 0 and 600. Due to network latency, values below 10 are advised
+                against.
+        Returns:
```

Comment:
Please document what to expect from the future and how to interpret it.

Reply:
Fixed.